packages/core/src/internal/motion/motion.service.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { MotionRegistry, TargetedAnimation } from './interfaces.js';
+
+const registry: MotionRegistry = {};
+
+const motionRegistry = new Proxy(registry, {
+  set: (target, key: string, value) => {
+    target[key] = value;
+    return true;
+  },
+});
+
+/**
+ *
+ * The ClarityMotion service is a static class that gives users the ability to retrieve
+ * and override animation configurations based on the Web Animations API that are targeted
+ * at a CSS selector defined in the configuration.
+ *
+ */
+export class ClarityMotion {
+  /**
+   * Returns a readonly reference of the registry of animations.
+   */
+  static get registry(): Readonly<MotionRegistry> {
+    return motionRegistry;
+  }
+
+  static has(name: string): boolean {
+    return !!name && !!motionRegistry[name];
+  }
+
+  static get(name: string): TargetedAnimation[] {
+    return motionRegistry[name] || [];
+  }
+
+  static add(animationName: string, animationConfig: TargetedAnimation[]) {
```

Comment:
ClarityIcons uses addIcon, since these (has & get) methods are similar should they method be more explicit? e.g  `addAnimation`

Reply:
`addIcon` is only `addIcon` because 3.0's API was backwards compatible with `@clr/icon`.

In the future, `addIcon` will become `add` and be deprecated. We just aren't in a hurry to do that and will likely keep it hanging around as an alias for `ClarityIcons.add` for the long term.